test/test_foreach.py

```diff
@@ -173,14 +184,71 @@ def test_add_list_error_cases(self, device):
             torch._foreach_add_(tensors1, tensors2)
 
     @dtypes(*torch.testing.get_all_dtypes())
-    def test_add_list_same_size(self, device, dtype):
-        tensors1 = [torch.zeros(10, 10, device=device, dtype=dtype) for _ in range(10)]
-        tensors2 = [torch.ones(10, 10, device=device, dtype=dtype) for _ in range(10)]
+    def test_bin_op_list(self, device, dtype):
+        if dtype == torch.bool:
+            return
+
+        tensors1 = [torch.zeros(20, 20, device=device, dtype=dtype) for _ in range(20)]
+        tensors2 = [torch.ones(20, 20, device=device, dtype=dtype) for _ in range(20)]
 
+        # add
         res = torch._foreach_add(tensors1, tensors2)
         torch._foreach_add_(tensors1, tensors2)
         self.assertEqual(res, tensors1)
-        self.assertEqual(res[0], torch.ones(10, 10, device=device, dtype=dtype))
+        self.assertEqual(tensors1, [torch.ones(20, 20, device=device, dtype=dtype) for _ in range(20)])
+
+        # sub
+        res = torch._foreach_sub(tensors1, tensors2)
```

Comment:
You could parameterize this function like

```
def _test(fn):
 ...
_test(torch._foreach_mul, torch.mul)
```
and then you have to write this code only once.

Reply:
The reason for ones and zeros is that we are testing all the dtypes and there is no random method that supports all. So im using ones and zeros to test that all ops are working as expected

Reply:
You still can create a helper function that produces a tensor of requested dtype. It could check dtype and depending on the dtype use either randint or rand. 
Right now the test flow is confusing, e.g. here you rely on `tensors1` being set to 0 by an inplace add test, but `add` and `sub` tests really should be independent, and parametrizing them as @cpuhrsch suggests would allow you to do that.

Reply:
Done